src/crystalline/classes/analysis/analysis.cr

```diff
@@ -59,13 +59,13 @@ module Crystalline::Analysis
 
     raise result if result.is_a? Exception
 
-    result.program.requires.each { |path|
-      diagnostics.try &.init_value("file://#{path}")
-    }
+    if !ignore_diagnostics
+      result.program.requires.each do |path|
+        diagnostics.try &.init_value("file://#{path}")
+      end
 
-    result.program.error_stack.try &.each do |e|
-      if e.is_a?(Crystal::TypeException) || e.is_a?(Crystal::SyntaxException)
-        diagnostics.append_from_exception(e) unless ignore_diagnostics
+      result.program.error_stack.try &.each do |e|
+        diagnostics.append_from_exception(e) if e.is_a?(Crystal::TypeException) || e.is_a?(Crystal::SyntaxException)
```

Comment:
```suggestion
        next unless e.is_a?(Crystal::TypeException) || e.is_a?(Crystal::SyntaxException)
        diagnostics.append_from_exception(e)
```

Reply:
I need to apply this suggestion manually (without just hit the github button :stuck_out_tongue: ), since the code snippet from suggested change is removing the line `result.program.error_stack.try &.each do |e|` by mistake.

Reply:
I'm ok with keeping the `if` one liner 😉.